dubbo-registry/dubbo-registry-nacos/src/main/java/org/apache/dubbo/registry/nacos/NacosRegistry.java

```diff
@@ -697,14 +737,21 @@ private NacosServiceName createServiceName(URL url) {
         return valueOf(url);
     }
 
-    private String getServiceName(URL url) {
+    private String getServiceName(URL url, boolean needCompatible) {
+        if (needCompatible) {
+            return getServiceName0(url, url.getCategory(DEFAULT_CATEGORY));
+        }
         return getServiceName(url, url.getCategory(DEFAULT_CATEGORY));
     }
 
     private String getServiceName(URL url, String category) {
         return category + SERVICE_NAME_SEPARATOR + url.getColonSeparatedKey();
     }
 
+    private String getServiceName0(URL url, String category) {
```

Comment:
I think `getCompatibleServiceName` would be easier to understand than `getServiceName0`. Please also use the `CompatibleXxx` naming pattern for other methods.